File: Codeforces/450B.cpp
#include <bits/stdc++.h>
#define IOS ios_base::sync_with_stdio(false); cin.tie(NULL);
#define loop(i,n) for(int i=0;i<n;i++)
#define mod 1000000007
#define for1(i,n) for(int i=1;i<=n;i++)
typedef long long int ll;
using namespace std;
void solve(){ 
   ll x,y,n,z,f;
	ll arr[7];
	cin>>x>>y>>n;
	arr[1]=(x+mod)%mod;
	arr[2]=(y+mod)%mod;
	arr[3]=(y-x+mod)%mod;
	arr[4]=(-x+mod)%mod;
	arr[5]=(-y+mod)%mod;
	arr[6]=(x-y+mod)%mod;
	if(n<=6) cout<<(arr[n]+mod)%mod;
	else{
		z=n%6;
		if(z==0) cout<<(arr[6]+mod)%mod;
		else cout<<(arr[z]+mod)%mod;
		
	}
}
int main() {
int t=1;
cin>>t;
while(t--){
 solve(); 
}
     return 0;
}

File: Codeforces/450B.py
mod = 1000000007
def solve(): 
  x,y = map(int,input().split())
  
  n = int(input())
  if(x==0 and y ==0):
    print(0)
    return
  arr = []
  arr.append(x%mod)
  arr.append(y%mod)
  i = 0
  w = 4
  while(i<w):
    m = len(arr)
    x = arr[m-1]%mod
    y = arr[m-2]%mod
    z = (x%mod - y%mod)%mod
    arr.append(z%mod)
    i+=1
  while(n>6):
    n=n%6 
  print(arr[n-1])

 
if __name__ == '__main__' :
  t = 1
  t= int(input())
  while(t>0):
     solve() 
     t-=1